src/plugins/intel_cpu/tests/unit/transformations/arm/grid_sample_decomposition.cpp

```diff
@@ -0,0 +1,417 @@
+    ov::element::Type grid_type;
+    bool align_corners;
+    ov::op::v9::GridSample::InterpolationMode interp_mode;
+    ov::op::v9::GridSample::PaddingMode padding_mode;
+};
+
+// ========== Base test classes ==========
+class GridSampleDecompositionStaticTest : public TransformationTestsF,
+                                          public WithParamInterface<GridSampleTestParams> {
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<GridSampleTestParams>& obj) {
+        const auto& p = obj.param;
+        std::ostringstream result;
+        result << "data_shape=" << p.data_shape
+               << "_grid_shape=" << p.grid_shape
+               << "_data_type=" << p.data_type
+               << "_grid_type=" << p.grid_type
+               << "_align=" << p.align_corners
+               << "_interp=";
+
+        switch (p.interp_mode) {
+            case ov::op::v9::GridSample::InterpolationMode::BILINEAR:
+                result << "bilinear";
+                break;
+            case ov::op::v9::GridSample::InterpolationMode::NEAREST:
+                result << "nearest";
+                break;
+            case ov::op::v9::GridSample::InterpolationMode::BICUBIC:
+                result << "bicubic";
+                break;
+        }
+
+        result << "_padding=";
+        switch (p.padding_mode) {
+            case ov::op::v9::GridSample::PaddingMode::ZEROS:
+                result << "zeros";
+                break;
+            case ov::op::v9::GridSample::PaddingMode::BORDER:
+                result << "border";
+                break;
+            case ov::op::v9::GridSample::PaddingMode::REFLECTION:
+                result << "reflection";
+                break;
+        }
+
+        return result.str();
+    }
+
+protected:
+    void SetUp() override {
+        TransformationTestsF::SetUp();
+        disable_rt_info_check();
+        const auto& p = GetParam();
+
+        auto data = std::make_shared<ov::op::v0::Parameter>(p.data_type, p.data_shape);
+        auto grid = std::make_shared<ov::op::v0::Parameter>(p.grid_type, p.grid_shape);
+
+        ov::op::v9::GridSample::Attributes attrs;
+        attrs.align_corners = p.align_corners;
+        attrs.mode = p.interp_mode;
+        attrs.padding_mode = p.padding_mode;
+
+        auto grid_sample = std::make_shared<ov::op::v9::GridSample>(data, grid, attrs);
+        auto result = std::make_shared<ov::op::v0::Result>(grid_sample);
+
+        model = std::make_shared<ov::Model>(ov::ResultVector{result}, ov::ParameterVector{data, grid});
+
+        // Create expected decomposed pattern using helper function
+        model_ref = create_expected_decomposed_pattern(p.data_shape, p.grid_shape, p.data_type, p.grid_type, attrs);
+
+        // Register the transformation to be tested
+        manager.register_pass<ov::intel_cpu::GridSampleDecomposition>();
+    }
+};
+
+class GridSampleDecompositionDynamicTest : public TransformationTestsF,
```

Comment:
It looks like we can easily combine `GridSampleDecompositionStaticTest` and `GridSampleDecompositionDynamicTest `